dockerworker/dockerworker.go

```diff
@@ -0,0 +1,70 @@
+// +build docker
+
+package dockerworker
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/taskcluster/taskcluster-client-go/tcqueue"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type DockerWorker struct {
+	LivelogWriter io.Writer
+	Logger        *log.Logger
+	TaskLogger    *log.Logger
+	Context       context.Context
+	Queue         *tcqueue.Queue
+	Client        *docker.Client
+}
+
+func New(ctx context.Context, queue *tcqueue.Queue, cli *docker.Client, taskID string, liveLogWriter io.Writer) *DockerWorker {
+	return &DockerWorker{
+		LivelogWriter: liveLogWriter,
+		Logger:        NewLogger(taskID),
+		TaskLogger:    NewTaskLogger(liveLogWriter),
+		Context:       ctx,
+		Queue:         queue,
+		Client:        cli,
+	}
+}
+
+// ValidatePayload validates the docker worker task payload
+func ValidatePayload(payload json.RawMessage) (result *gojsonschema.Result, err error) {
+	const schema = "https://raw.githubusercontent.com/taskcluster/docker-worker/master/schemas/v1/payload.json"
```

Comment:
We cannot download from this github url for every task that runs in production. The worker should not depend on github.com at runtime, and the load would be enormous. Also it is pointing to a moving target, using the master branch.

Reply:
Oh, I put this as temporary and forgot it, thanks for pointing this out.